dspace-uclouvain/src/main/java/org/dspace/uclouvain/submissionMetadataGenerators/generators/DegreeMetadataGenerator.java

```diff
@@ -49,59 +48,36 @@ public String getGeneratorName() {
             List<MetadataValue> currentDegreeCodes = currentItemService.getMetadata(
-                item,
-                this.degreeCodeField.getSchema(),
+                item, this.degreeCodeField.getSchema(),
                 this.degreeCodeField.getElement(),
                 this.degreeCodeField.getQualifier(),
                 null
             );
-            // Generate code and labels in a list of pairs: [['code1', 'label1'], ['code2', 'label2'], ['code3', 'label3']]
-            List<List<String>> codeAndLabels = this.generateCodeAndLabels(currentDegreeCodes);
-            // Loop over the pairs and add the corresponding metadata to the item
-            for (Integer elementIndex = 0; elementIndex < codeAndLabels.size(); elementIndex++) {
-                List<String> values = codeAndLabels.get(elementIndex);
-                // Replace degree code
-                currentItemService.replaceSecuredMetadata(
-                    ctx, item,
-                    this.degreeCodeField.getSchema(),
-                    this.degreeCodeField.getElement(),
-                    this.degreeCodeField.getQualifier(),
-                    null, values.get(0), null, -1, elementIndex, 0
-                );
-                // Add degree label
-                currentItemService.addMetadata(
-                    ctx, item,
-                    this.degreeLabelField.getSchema(),
-                    this.degreeLabelField.getElement(),
-                    this.degreeLabelField.getQualifier(),
-                    null, values.get(1), null, -1, 0
-                );
-            }
-
-            // 3) Generate new information based on the distinct retrieved degree codes and the degree mappers file
-            List<String> distinctCodes = codeAndLabels.stream().map(x -> x.get(0)).distinct().collect(Collectors.toList());
+            List<String> distinctCodes = currentDegreeCodes
+                .stream()
+                .map(MetadataValue::getValue)
+                .distinct()
+                .collect(Collectors.toList());
             List<DegreeMapper> mappings = this.degreeMappersService.getDegreeMappersForDegreeCodes(distinctCodes);
             // For each mapping found, add additional info to the item
             for (DegreeMapper mapping: mappings) {
                 generateAdditionalMetadataFromMapper(ctx, item, currentItemService, mapping);
             }
-
-            // 4) Update the item with the additional metadata
+            // 3) Update the item with the additional metadata
             currentItemService.update(ctx, item);
 
         } catch (SQLException e) {
             throw new GeneratorProcessException("An error occurred while replacing degree metadata", e);
-        } catch (MalformedDegreeCodeException e) {
-            throw new GeneratorProcessException("Could not generate degree code and label due to malformed degree code", e);
```

Comment:
We could also remove the 'MalformedDegreeCodeException' class since we are not handling the splitting anymore.